client/components/data/query-site-verticals/index.jsx

```diff
@@ -0,0 +1,73 @@
+import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import { requestVerticals } from 'calypso/state/site-verticals/actions';
+import { getVerticals } from 'calypso/state/site-verticals/selectors';
+
+export class QuerySiteVerticals extends Component {
```

Comment:
As we have introduced `react-query`, we could just make `useSiteVerticals` hook and use `react-query` to fetch the verticals list

Related
* p7H4VZ-34y-p2
* p4TIVU-9JT-p2
* https://github.com/Automattic/wp-calypso/tree/trunk/client/data